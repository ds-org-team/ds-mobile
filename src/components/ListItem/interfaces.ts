import { ReactNode } from 'react';
import { ImageXorLabel } from '../Avatar/interfaces';

export type ListItemProps = {
  rightComponent?: ReactNode;
  title: string;
  text?: string;
  icon?: string;
  avatar?: ImageXorLabel;
  tags?: string[];
  bottomComponent?: ReactNode;
};
