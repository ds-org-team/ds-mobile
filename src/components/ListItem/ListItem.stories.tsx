import React from 'react';
import { withDesign } from 'storybook-addon-designs';
import ListItem from './ListItem';

export default {
  title: 'Componente/ListItem',
  component: ListItem,
  argTypes: {
    title: {
      control: { type: 'text' },
      defaultValue: 'Item',
    },
    icon: {
      control: {
        type: 'text',
      },
      defaultValue: 'account',
    },
  },
  decorators: [withDesign],
};

const urlHandoff =
  'https://www.figma.com/file/3raVfIADTUZCzFOOaQ9PMQ/HANDOFF-%7C-Core-Components-Mobile-%7C-Institucional?node-id=228%3A5457';
interface ListItemProps {
  title: string;
  icon: string;
}

export const Primary = ({ title, icon }: ListItemProps): React.ReactNode =>
(
  <ListItem title={title} icon={icon} />
);

Primary.parameters = {
  design: {
    type: 'figma',
    url: urlHandoff,
  },
};
