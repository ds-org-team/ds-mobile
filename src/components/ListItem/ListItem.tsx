import React from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import Box from '../Box';
import { ListItemProps } from './interfaces';
import Text from '../Text';

const ListItem: React.FC<ListItemProps> = ({ title, icon, ...props }) => (
  <Box
    minHeight="nano"
    backgroundColor="neutral-dark"
    flexDirection="row"
    {...props}
    py="nano"
    px="nano"
  >
    <Box mr="nano" borderRadius="circular">
      {icon && <Icon name={icon} size={16} />}
    </Box>
    <Text fs="md">{title}</Text>
  </Box>
);

export default ListItem;
