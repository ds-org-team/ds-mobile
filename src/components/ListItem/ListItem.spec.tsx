import React from 'react';
import { cleanup, render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import themeMaestro from '../../themes/maestro';
import ListItem from './ListItem';

afterEach(cleanup);

it('should render ListItem component', async () => {
  const { toJSON } = render(
    <ThemeProvider theme={themeMaestro}>
      <ListItem />
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});
