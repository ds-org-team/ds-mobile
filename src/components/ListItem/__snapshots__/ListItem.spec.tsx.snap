// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Text
  style={
    Array [
      Object {
        "display": "flex",
        "fontFamily": "\\"Inter\\", sans-serif",
        "fontWeight": "400",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  Hello from ListItem!
</Text>
`;

exports[`should render ListItem component 1`] = `
<Text
  style={
    Array [
      Object {
        "display": "flex",
        "fontFamily": "\\"Inter\\", sans-serif",
        "fontWeight": "400",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  Hello from ListItem!
</Text>
`;
